src/Infrastructure/BotSharp.Abstraction/Conversations/ConversationHookProvider.cs

```diff
@@ -0,0 +1,20 @@
+namespace BotSharp.Abstraction.Conversations;
+
+public class ConversationHookProvider
+{
+    public IEnumerable<IConversationHook> Hooks { get; }
+
+    private readonly Lazy<IEnumerable<IConversationHook>> _hooksOrderByPriority;
+
+    public IEnumerable<IConversationHook> HooksOrderByPriority
+        => _hooksOrderByPriority.Value;
+
+    public ConversationHookProvider(IEnumerable<IConversationHook> conversationHooks)
+    {
+        Hooks = conversationHooks;
```

Comment:
When will `conversationHooks` be populated by all the services of `IConversationHook`?